src/Config.php

```diff
@@ -128,45 +154,49 @@ class_exists('\Illuminate\Support\Str') => [
                 ],
                 'paths' => [],
             ],
-        ], JSON_PRETTY_PRINT));
+        ];
     }
 
     /**
-     * Adds a word to the ignore list.
+     * Writes the json config structure to the config file (usually peck.json).
      *
-     * @param  array<int, string>  $words
+     * @param  non-empty-array<'ignore'|'preset', array{words: array<int, string>, paths: array<int, string>}|string>  $config
      */
-    public function ignoreWords(array $words): void
+    private static function writeConfigFile(array $config): bool
     {
-        $this->whitelistedWords = array_merge($this->whitelistedWords, array_map(strtolower(...), $words));
+        $filePath = self::getResolvedFilePath();
 
-        $this->persist();
+        if (file_exists($filePath)) {
+            return false;
+        }
+
+        return (bool) file_put_contents($filePath, json_encode($config, JSON_PRETTY_PRINT));
     }
 
     /**
-     * Checks if the word is ignored.
+     * returns the resolved config file path when a closure is available, or returns the default file in the project path
      */
-    public function isWordIgnored(string $word): bool
+    private static function getResolvedFilePath(): string
     {
-        return in_array(strtolower($word), [
-            ...$this->whitelistedWords,
-            ...PresetProvider::whitelistedWords($this->preset),
-        ]);
+        return self::$resolveConfigFilePathUsing instanceof Closure
+            ? (self::$resolveConfigFilePathUsing)()
+            : sprintf('%s/%s', ProjectPath::get(), self::JSON_CONFIGURATION_NAME);
     }
 
     /**
      * Save the configuration to the file.
      */
     private function persist(): void
     {
-        $filePath = ProjectPath::get().'/'.self::JSON_CONFIGURATION_NAME;
-
-        file_put_contents($filePath, json_encode([
-            ...$this->preset !== null ? ['preset' => $this->preset] : [],
-            'ignore' => [
-                'words' => $this->whitelistedWords,
-                'paths' => $this->whitelistedPaths,
-            ],
-        ], JSON_PRETTY_PRINT));
+        self::writeConfigFile(array_merge(
+            self::defaultConfigStructure(),
```

Comment:
Why do we merge the default structure here? We didn't do this before the refactoring.